sync/src/synchronizer/mod.rs

```diff
@@ -734,6 +721,23 @@ impl<CS: ChainStore> CKBProtocolHandler for Synchronizer<CS> {
         self.process(nc.as_ref(), peer_index, msg);
     }
 
+    fn connected(&mut self, nc: Box<CKBProtocolContext>, peer_index: PeerIndex, _version: &str) {
+        info!(target: "sync", "SyncProtocol.connected peer={}", peer_index);
+        self.on_connected(nc.as_ref(), peer_index);
+    }
+
+    fn disconnected(&mut self, _nc: Box<CKBProtocolContext>, peer_index: PeerIndex) {
+        info!(target: "sync", "SyncProtocol.disconnected peer={}", peer_index);
+        let mut state = self.peers.state.write();
+        if let Some(peer_state) = state.get(&peer_index) {
+            if peer_state.sync_started && self.n_sync.fetch_sub(1, Ordering::Release) == 0 {
+                panic!("Synchronizer n_sync overflow");
```

Comment:
could you please explain in more detail when will panic?

Reply:
if our code running as we expect, it shouldn't happen.
 ` fetch_sub` wraps around on overflow, we need check manually, panic here to prevent some bug be hidden silently.